design/read-only-registry.md

```diff
@@ -0,0 +1,174 @@
+Other command line arguments can be used for headless and/or unattended install experience without any additional changes.
+
+### Updates
+
+#### Online Admin Console
+
+Upgrading Admin Console installed in online mode does not change.
+
+#### Airgapped Admin Console
+
+Upgrading Admin Console installed in airgapped mode has to be done with the `--disable-image-push` flag specified on the command line:
+
+```
+kubectl kots admin-console upgrade \
+  --kotsadm-registry my.private.registry/myapp \
+  --registry-username <username> \
+  --registry-password <password> \
+  --disable-image-push
+```
+
+In this case, the `kots admin-console push-images` command can be ommitted as well.
+
+#### Online application without private registry
+
+Upgrading online application without private registry does not change.
+
+#### Online application with private registry
+
+Upgrading online application with private registry does not change from user's perspective.
+Admin Console will still craete a new version downlaoded from the server and rewrite all images to be pulled from the private registry.
+
+Internally the functionality changes as follows:
+- Images will not be pushed to private registry.
+
+#### Airgapped application (UI)
+
+Upgrading airgapped application using Admin Console UI does not change from user's perspective.
+Airgap bundle still has to be uploaded through the UI.
+Admin Console will still create a new application version and rewrite all images to be pulled from the private registry.
+
+Internally the functionality changes as follows:
+- Images will not be pushed to private registry.
+
+#### Airgapped application (CLI)
+
+Upgrading airgapped application using Admin Console CLI does not change from user's perspective.
+Admin Console will still create a new application version and rewrite all images to be pulled from the private registry.
+
+```
+kubectl kots upstream upgrade app-slug \
+  --airgap-bundle /path/to/app.airgap \
+  --kotsadm-namespace myapp \
+  --kotsadm-registry my.private.registry \
+  --registry-username <username> \
+  --registry-password <password> \
+  --disable-image-push
+  ```
+
+Internally the functionality changes as follows:
+- Images will not be pushed to private registry.
+
+### Registry Settings page
+
+Registry settings page will have a checkbox named `Disable Image Push`.
+
+#### Online application installs
+
+In online installs this checkbox is can be toggled freely.
+
+- When the checkbox is unchecked, saving registry works without any changes.
+- When checkbox is checked, saving registry settings will not push images.
+
+#### Airgapped installs
+
+In airgapped environments, changing registry host and namespace is not supported today.
+Therefore this checkbox will reflect the value of the `--disable-image-push` flag used to install latest app version.
+
+### Preflight checks
+
+A new preflight check will be added to validate that images are present in their respective registries.
```

Comment:
@markpundsack I that we will can create a special preflight check.  Vendor would have to choose to add it.

Reply:
Hmm... since the vendor isn't really part of the decision to use read-only registry or not, it doesn't make sense to rely on the vendor choosing to add the preflight check. Seems to me like we should inject that when necessary based on end-user enterprise choices.